### PR TITLE
[IMP] mail: introduce 'FollowerSubtypeView' model

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2793280

### CODE REVIEW

File: addons/mail/static/src/components/follower_subtype/follower_subtype.js [hunk 1]
```diff
@@ -48,6 +55,7 @@ Object.assign(FollowerSubtype, {
     props: {
         followerLocalId: String,
         followerSubtypeLocalId: String,
```
Comment (seb-odoo) [inline addons/mail/static/src/components/follower_subtype/follower_subtype.js:57]: Those were identifying no? Should no longer be necessary and use the new relation instead.

Comment (alexkuhn) [reply addons/mail/static/src/components/follower_subtype/follower_subtype.js:57]: I make the changes in other following PRs:
https://github.com/odoo/odoo/pull/87014
https://github.com/odoo/odoo/pull/87013

It's not done in this PR to keep PR as small as possible while keeping working code.

**Review by phenix-factory — APPROVED** [review]
I can't find anything. I don't like that the test is fully based on partner following other partner, it make it harder to read for me. But that feel out of scope.
LGTM.

**Comment by alexkuhn** [comment]
> I can't find anything. I don't like that the test is fully based on partner following other partner, it make it harder to read for me. But that feel out of scope.
> LGTM.

Agree, the test before change cheat by making models by hand, big assumption on proper setup for so-so working code.
It easily broke with introduction of `FollowerSubtypeView`.
Here just made as few adjustments I could by relying on `this.data`.
Tests are not nice but it does the job for the time being (makes runbot green while keeping same coverage and more robust test, at cost of test readability)

**Comment by alexkuhn** [comment]
force-push is just for rebase, the ci/runbot UI was stuck on github, and I suspect it could prevent robodoo to r-plus, see https://github.com/odoo/odoo/pull/86815#issuecomment-1076432270)

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**